var/spack/repos/builtin/packages/openssl/package.py

```diff
@@ -172,5 +172,13 @@ def link_system_certs(self):
                 os.rmdir(pkg_certs)
                 os.symlink(sys_certs, pkg_certs)
 
+    def patch(self):
+        if self.spec.satisfies('%nvhpc'):
+            # Remove incompatible preprocessor flags
+            filter_file('-MF ', '',
+                        'Configurations/unix-Makefile.tmpl', string=True)
+            filter_file('-MT \$\@ ', '',
```

Comment:
```suggestion
            filter_file(r'-MT \$\@ ', '',
```
This will fix flake8